packages/admin-next/dashboard/src/v2-routes/shipping/service-zone-areas-edit/components/edit-region-areas-form/edit-service-zone-areas-form.tsx

```diff
@@ -0,0 +1,346 @@
+  Alert,
+  Badge,
+  Button,
+  Checkbox,
+  Heading,
+  IconButton,
+  Text,
+  toast,
+} from "@medusajs/ui"
+import { RegionCountryDTO, RegionDTO, ServiceZoneDTO } from "@medusajs/types"
+import { useTranslation } from "react-i18next"
+import { XMarkMini } from "@medusajs/icons"
+
+import {
+  RouteFocusModal,
+  useRouteModal,
+} from "../../../../../components/route-modal"
+import { SplitView } from "../../../../../components/layout/split-view"
+import {
+  useCreateServiceZone,
+  useUpdateServiceZone,
+} from "../../../../../hooks/api/stock-locations"
+import { useEffect, useMemo, useState } from "react"
+import { useCountryTableQuery } from "../../../../regions/common/hooks/use-country-table-query"
+import { useCountries } from "../../../../regions/common/hooks/use-countries"
+import { countries as staticCountries } from "../../../../../lib/countries"
+import { useDataTable } from "../../../../../hooks/use-data-table"
+import { useCountryTableColumns } from "../../../../regions/common/hooks/use-country-table-columns"
+import { DataTable } from "../../../../../components/table/data-table"
+
+const PREFIX = "ac"
+const PAGE_SIZE = 50
+
+const ConditionsFooter = ({ onSave }: { onSave: () => void }) => {
+  const { t } = useTranslation()
+
+  return (
+    <div className="flex items-center justify-end gap-x-2 border-t p-4">
+      <SplitView.Close type="button" asChild>
+        <Button variant="secondary" size="small">
+          {t("actions.cancel")}
+        </Button>
+      </SplitView.Close>
+      <Button size="small" type="button" onClick={onSave}>
+        {t("actions.select")}
+      </Button>
+    </div>
+  )
+}
+
+const EditeServiceZoneSchema = zod.object({
+  countries: zod.array(zod.string().length(2)).min(1),
+})
+
+type EditServiceZoneAreasFormProps = {
+  fulfillmentSetId: string
+  locationId: string
+  zone: ServiceZoneDTO
+}
+
+export function EditServiceZoneAreasForm({
+  fulfillmentSetId,
+  locationId,
+  zone,
+}: EditServiceZoneAreasFormProps) {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+  const [open, setOpen] = useState(false)
+  const [rowSelection, setRowSelection] = useState<RowSelectionState>(
+    zone.geo_zones
+      .map((z) => z.country_code)
+      .reduce((acc, v) => {
+        acc[v] = true
+        return acc
+      }, {})
+  )
+
+  const form = useForm<zod.infer<typeof EditeServiceZoneSchema>>({
+    defaultValues: {
+      countries: zone.geo_zones.map((z) => z.country_code),
+    },
+    resolver: zodResolver(EditeServiceZoneSchema),
+  })
+
+  const { mutateAsync: editServiceZone, isPending: isLoading } =
+    useUpdateServiceZone(fulfillmentSetId, zone.id, locationId)
+
+  const handleSubmit = form.handleSubmit(async (data) => {
+    try {
+      await editServiceZone({
+        geo_zones: data.countries
+          .sort((a, b) => a.localeCompare(b))
+          .map((iso2) => ({
+            country_code: iso2,
+            type: "country",
+          })),
```

Comment:
Good catch - I wanted to sort the countries list since it's just a good UX practice but this should sort by name instead of country code. Will push a fix.